### PR TITLE
[ios] Hint for RouteTo button to make routes without Point2Point tab

### PR DESCRIPTION
**Pull request by kirylkaveryn**
Closes https://github.com/organicmaps/organicmaps/issues/6256

## Problem
The point-to-point button was removed, so it needs to educate users to build routes.

## Solving
In this PR as mentioned in https://github.com/organicmaps/organicmaps/issues/6256 was added:
1. tint for RouteTo tab when the user tap on some location
2. red badge to the "?" (help) tab that routes the user to the FAQ screen
Both hints **appear once** only for:
- the first app launch after installation 
or 
- the first launch after the app is updated from the previous version to the current

Simulator Screen Recording - iPhone 14 Pro - 2023-10-31 at 23 43 28

### CODE REVIEW

File: iphone/Maps/Classes/FirstSession.mm [hunk 1]
```diff
@@ -55,11 +55,14 @@ of this software and associated documentation files (the "Software"), to deal
...
 static NSString * const kTotalSecondsInTheApp = @"AlohalyticsTotalSecondsInTheApp";
 static NSString * const kIsAlohalyticsDisabledKey = @"AlohalyticsDisabledKey";
+static NSString * const kCFBundleShortVersionStringKey = @"CFBundleShortVersionString";
+static NSString * const kUDLastAppVersionPromptedForReview = @"kUDLastVersionPromptedForReview";
 
 // Used to calculate session length and total time spent in the app.
 // setup should be called to activate counting.
 static NSDate * gSessionStartTime = nil;
-static BOOL gIsFirstSession = NO;
+static BOOL gIsFirstSession = false;
```
Comment (biodranik) [inline iphone/Maps/Classes/FirstSession.mm:64]: What's the motivation for this change?
For C++ true/false you need to use C++ `bool` type.

File: iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m [hunk 2]
```diff
@@ -182,4 +190,14 @@ - (void)setupBookmarkButton:(BOOL)isSelected {
   animationIV.animationRepeatCount = 1;
 }
 
+- (BOOL)needsToHighlightRouteToButton {
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  if ((FirstSession.isFirstSession || FirstSession.isFirstSessionInCurrentVersion)
+    && ![userDefaults boolForKey:kUDDidHighlightRouteToButton]) {
+    [userDefaults setBool:true forKey:kUDDidHighlightRouteToButton];
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m:197]: Is this approach reliable when PP is recreated on a second click?

A more robust solution is to highlight it until it is not pressed the first time.

File: iphone/Maps/Classes/FirstSession.mm [hunk 3]
```diff
@@ -92,6 +101,22 @@ + (BOOL)isFirstSession {
   return [self installationId] != nil && gIsFirstSession;
 }
 
++ (BOOL)isFirstSessionInCurrentVersion {
+  return gIsFirstSessionInCurrentVersion;
```
Comment (biodranik) [inline iphone/Maps/Classes/FirstSession.mm:105]: There are several potential flaws with this approach. Let's simplify it for now and always highlight ? and Route To until they are pressed by a user.

Comment (kirylkaveryn) [reply iphone/Maps/Classes/FirstSession.mm:105]: Tint tab on every opening of PP screen?

File: iphone/Maps/Classes/FirstSession.mm (hunk 1 above)
Comment (kirylkaveryn) [reply iphone/Maps/Classes/FirstSession.mm:64]: I'll set both NO

File: iphone/Maps/Classes/FirstSession.mm (hunk 3 above)
Comment (biodranik) [reply iphone/Maps/Classes/FirstSession.mm:105]: A user launches the app update or the new install, sees it tinted (blue) and sees a * near ?.
A user presses tinted button Route To (that means that user has discovered how to build routes). Tint disappears forever.
A user presses ? button. * disappears. It is possible that * will appear again next time for another feature/announcement.

File: iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m (hunk 2 above)
Comment (kirylkaveryn) [reply iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m]: Refactor this to:
`return  ![NSUserDefaults.standardUserDefaults boolForKey:kUDDidHighlightRouteToButton] ;`
So it works the same manner but without FirstSession

File: data/faq.html [hunk 4]
```diff
@@ -383,2403 +383,2403 @@ <h4 lang="uk">
...
+          options.
         </p>
+      </dd>
 
+      <dd lang="ru">
         <p>
-          Чем больше небезразличных людей, тем точнее будут карты для всех нас.
-          Мы в команде считаем, что подробные карты всей нашей планеты – это
-          всего лишь вопрос времени.
+          Убедитесь, что ваше местоположение определено и выберите пункт
```
Comment (biodranik) [inline data/faq.html:444]: There are so many edits in this file, so it is impossible to review it.

I propose not to change it in this PR and create a separate one, where the first commit will have formatting fixes (if they are really necessary), and the second commit will have actual changes.

File: iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift [hunk 5]
```diff
@@ -56,9 +56,14 @@ class BottomTabBarViewController: UIViewController {
   @IBAction func onSearchButtonPressed(_ sender: Any) {
     presenter.onSearchButtonPressed()
   }
-
+  
   @IBAction func onHelpButtonPressed(_ sender: Any) {
-    presenter.onHelpButtonPressed()
+    guard helpBadge.isHidden else {
```
Comment (biodranik) [inline iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift:61]: Why guard/return is better than a simple if/else?

File: iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift [hunk 6]
```diff
@@ -1,3 +1,6 @@
+
+private let kUDDidShowHelpButtonEducationalHint = "kUDDidShowHelpButtonEducationalHint"
```
Comment (biodranik) [inline iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift:2]: Let's name it something like "first time routing educational hiht". We likely provide other hints too for the help button.

File: data/faq.html (hunk 4 above)
Comment (kirylkaveryn) [reply data/faq.html]: removed html from PR

File: iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift (hunk 5 above)
Comment (kirylkaveryn) [reply iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift]: Just trying to avoid additional scope for this super rare case with badge is visible. 
Fixed to if/else!

File: iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift (hunk 6 above)
Comment (kirylkaveryn) [reply iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift]: Fixed!

File: iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift [hunk 7]
```diff
@@ -56,9 +56,14 @@ class BottomTabBarViewController: UIViewController {
   @IBAction func onSearchButtonPressed(_ sender: Any) {
     presenter.onSearchButtonPressed()
   }
-
+
```
Comment (biodranik) [inline iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift:59]: Unnecessary spaces?

File: iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m [hunk 8]
```diff
@@ -106,6 +110,9 @@ - (void)configButton:(BOOL)isSelected enabled:(BOOL)isEnabled {
       break;
     case MWMActionBarButtonTypeRouteTo:
       [self.button setImage:[UIImage imageNamed:@"ic_route_to"] forState:UIControlStateNormal];
+      if ([self needsToHighlightRouteToButton]) {
+        self.button.coloring = MWMButtonColoringBlue;
+      }
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m:115]:
```suggestion
      if ([self needsToHighlightRouteToButton])
        self.button.coloring = MWMButtonColoringBlue;
```

File: iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m [hunk 9]
```diff
@@ -182,4 +192,12 @@ - (void)setupBookmarkButton:(BOOL)isSelected {
   animationIV.animationRepeatCount = 1;
 }
 
+- (BOOL)needsToHighlightRouteToButton {
+  return ![NSUserDefaults.standardUserDefaults boolForKey:kUDDidHighlightRouteToButton];
+}
+
+- (void)setRouteToButtonDidHighlighted:(BOOL)highlighted {
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/PlacePageLayout/ActionBar/MWMActionBarButton.m:199]:
```suggestion
- (void)disableRouteToButtonHighlight {
```

**Review by biodranik — APPROVED** [review]
LGTM with a few comments.

File: iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift (hunk 7 above)
Comment (kirylkaveryn) [reply iphone/Maps/UI/BottomMenu/TabBar/BottomTabBarViewController.swift:59]: I suppose it is Ok if I'll remove all Unnecessary spaces in the file?
image

**Review by biodranik — APPROVED** [review]
Thank you!
